### PR TITLE
fix: eliminate warning about flet being an obsolete macro

### PR DESCRIPTION
**Pull request by crimeminister**
Replaces usage of `flet` by `cl-flet` in response to the following warning on startup:
```
‘flet’ is an obsolete macro (as of 24.3); use either ‘cl-flet’ or ‘cl-letf’.
```

### CODE REVIEW

**Review by jcs090218 — APPROVED** [review]
I don't have the push permission but this LGTM.

**Comment by tarsius** [comment]
This replaces one obsolete `cl` function with the respective `cl-lib` function, but without requiring that library and without replacing all the other obsolete functions accordingly. I don't think doing just that makes sense. If you update this properly, then also adjust the whitespace where necessary (while making sure you keep using tabs or spaces; whatever is currently being used). Note that emacs 28 comes with a function to do the conversion (I don't remember its name).

**Comment by jcs090218** [comment]
True, we will need to change all `cl` functions to `cl-lib`'s version. And change `(require 'cl)` to `(require 'cl-lib)`.

**Comment by tarsius** [comment]
This is what I had in mind: 4e3376849aeea894e235f125697f088f22b4c2d6.

**Comment by crimeminister** [comment]
Thanks for taking a look @tarsius.

**Pull request closed without merging**
